src/tests/Microsoft.PowerFx.Interpreter.Tests.Shared/DependencyTests.cs

```diff
@@ -0,0 +1,223 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.IR.Nodes;
+using Microsoft.PowerFx.Core.Tests.AssociatedDataSourcesTests;
+using Microsoft.PowerFx.Core.Tests.Helpers;
+using Microsoft.PowerFx.Core.Texl;
+using Microsoft.PowerFx.Core.Texl.Builtins;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Interpreter;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Core.Tests
+{
+    public class DependencyTests : PowerFxTest
+    {
+        [Theory]
+        [InlineData("1+2", "")] // none
+        [InlineData("ThisRecord.'Address 1: City' & 'Account Name'", "Read Accounts: address1_city, name;")] // basic read
+
+        [InlineData("numberofemployees%", "Read Accounts: numberofemployees;")] // unary op
+        [InlineData("ThisRecord", "Read Accounts: ;")] // whole scope
+        [InlineData("{x:5}.x", "")]
+        [InlineData("With({x : ThisRecord}, x.'Address 1: City')", "Read Accounts: address1_city;")] // alias
+        [InlineData("With({'Address 1: City' : \"Seattle\"}, 'Address 1: City' & 'Account Name')", "Read Accounts: name;")] // 'Address 1: City' is shadowed
+        [InlineData("With({'Address 1: City' : 5}, ThisRecord.'Address 1: City')", "")] // shadowed
+        [InlineData("LookUp(local,'Address 1: City'=\"something\")", "Read Accounts: address1_city;")] // Lookup and RowScope
+        [InlineData("Filter(local,numberofemployees > 200)", "Read Accounts: numberofemployees;")]
+        [InlineData("First(local)", "Read Accounts: ;")]
+        [InlineData("First(local).'Address 1: City'", "Read Accounts: address1_city;")]
+        [InlineData("Last(local)", "Read Accounts: ;")]
+        [InlineData("local", "Read Accounts: ;")] // whole table
+        [InlineData("12 & true & \"abc\" ", "")] // walker ignores literals
+        [InlineData("12;'Address 1: City';12", "Read Accounts: address1_city;")] // chaining
+        [InlineData("ParamLocal1.'Address 1: City'", "Read Accounts: address1_city;")] // basic read
+
+        // Basic scoping
+        [InlineData("Min(local,numberofemployees)", "Read Accounts: numberofemployees;")]
+        [InlineData("Average(local,numberofemployees)", "Read Accounts: numberofemployees;")]
+
+        // Patch
```

Comment:
Test: 
Patch(local, First(local), { field1 : First(local).field2  } )

local.field1 is a write. 
local.field2 is a read

Reply:
Test case added.

Reply:
This test result is now different after our internal chat.
Result: `Entity local: field1, field2`.